pkg/providers/ovirt/mapper/mapper.go

```diff
@@ -0,0 +1,200 @@
+package mapper
+
+import (
+	"strconv"
+
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	provider "github.com/kubevirt/vm-import-operator/pkg/providers"
+	"github.com/kubevirt/vm-import-operator/pkg/utils"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+const (
+	cdiAPIVersion = "cdi.kubevirt.io/v1alpha1"
+)
+
+var biosMapping = map[string]*kubevirtv1.Bootloader{
+	"q35_sea_bios":    &kubevirtv1.Bootloader{BIOS: &kubevirtv1.BIOS{}},
+	"q35_secure_boot": &kubevirtv1.Bootloader{BIOS: &kubevirtv1.BIOS{}},
+	"q35_ovmf":        &kubevirtv1.Bootloader{EFI: &kubevirtv1.EFI{}},
+}
+
+var archMapping = map[string]string{
+	"x86_64": "q35",
+}
+
+// OvirtMapper is struct that holds attributes needed to map oVirt VM to kubevirt VM
+type OvirtMapper struct {
+	vm        *ovirtsdk.Vm
+	mappings  *v2vv1alpha1.OvirtMappings
+	creds     provider.DataVolumeCredentials
+	namespace string
+}
+
+// NewOvirtMapper create ovirt mapper object
+func NewOvirtMapper(vm *ovirtsdk.Vm, mappings *v2vv1alpha1.OvirtMappings, creds provider.DataVolumeCredentials, namespace string) *OvirtMapper {
+	return &OvirtMapper{
+		vm:        vm,
+		mappings:  mappings,
+		creds:     creds,
+		namespace: namespace,
+	}
+}
+
+// MapVM map oVirt API VM definition to kubevirt VM definition
+func (o *OvirtMapper) MapVM(targetVMName *string) *kubevirtv1.VirtualMachine {
+	vmSpec := kubevirtv1.VirtualMachine{
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec: kubevirtv1.VirtualMachineSpec{
+			Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+				Spec: kubevirtv1.VirtualMachineInstanceSpec{
+					Domain: kubevirtv1.DomainSpec{},
+				},
+			},
+		},
+	}
+
+	// Map name
+	vmSpec.ObjectMeta.Name, _ = o.resolveVMName(targetVMName, o.vm)
+
+	// Map CPU
+	vmSpec.Spec.Template.Spec.Domain.CPU = o.mapCPU()
+
+	// Map bios
+	vmSpec.Spec.Template.Spec.Domain.Firmware = o.mapFirmware()
+
+	// Map machine type
+	vmSpec.Spec.Template.Spec.Domain.Machine = *o.mapArchitecture()
+
+	return &vmSpec
+}
+
+// MapDisks map the oVirt VM disks to the map of CDI DataVolumes specification, where
+// map id is the id of the oVirt disk
+func (o *OvirtMapper) MapDisks() map[string]cdiv1.DataVolume {
+	diskAttachments, _ := o.vm.DiskAttachments()
+	dvs := make(map[string]cdiv1.DataVolume, len(diskAttachments.Slice()))
+
+	for _, diskAttachment := range diskAttachments.Slice() {
+		attachID, _ := diskAttachment.Id()
+		disk, _ := diskAttachment.Disk()
+		diskID, _ := disk.Id()
+		sdClass := o.getStorageClassForDisk(disk, o.mappings)
+		quantity, _ := resource.ParseQuantity(strconv.FormatInt(disk.MustProvisionedSize(), 10))
```

Comment:
We could get a panic from `MustProvisionedSize`

Reply:
Right, changed.